### PR TITLE
add specific error classes for SDKClient and MirrorNodeCLient

### LINKED ISSUE #213: Add client error classes to streamline error handling
### Problem

Currently both clients (sdk and mirrorNode) can send a variation of errors.
These leak info, make logs large and also vary in nature so it's hard to sometime differentiate and infrom how the `eth_x` calls should handle them.

### Solution

We should add a `SDKClientError` and `MirrorNodeClientError` class that consumes the returned errors and captures the value information e.g. type, status code etc.

This should be thrown by the clients, checked for by `eth` class and managed as needed.
This way server logs and details are very small and specific and don't leak `eth` or client info

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by lukelee-sl**
Signed-off-by: lukelee-sl <luke.lee@swirldslabs.com>

**Description**:
Create error classes for SDKClient and MirrorNodeClient classes.

**Related issue(s)**:

Fixes #213 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by ar-conmit — DISMISSED** [review]
LG

File: packages/relay/src/lib/errors.ts [hunk 1]
```diff
@@ -107,3 +109,27 @@ export const predefined = {
...
   }),
 };
+
+export class SDKClientError extends Error {
+  public statusCode: number;
+
+  constructor(message: string, statusCode?: number) {
+    super(message);
+    this.statusCode = statusCode ? statusCode : Status.Unknown._code;
+    this.name = "SDKClientError";
```
Comment (Nana-EC) [inline packages/relay/src/lib/errors.ts:119]: Not sure this is needed since it's implied by the type.

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Nice start.
Let's move some of the logic into the error class files

File: packages/relay/src/lib/errors/JsonRpcError.ts [hunk 2]
```diff
@@ -106,4 +106,4 @@ export const predefined = {
     code: -32602,
     message: 'Value below 10_000_000_000 wei which is 1 tinybar'
   }),
-};
+};
```
Comment (Nana-EC) [inline packages/relay/src/lib/errors/JsonRpcError.ts:109]: restore new line.
Make git diff easier

File: packages/relay/src/lib/errors/MirrorNodeClientError.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+export class MirrorNodeClientError extends Error {
+    public statusCode: number;
+  
+    constructor(message: string, statusCode: number) {
+      super(message);
+      this.statusCode = statusCode;
+  
+      Object.setPrototypeOf(this, MirrorNodeClientError.prototype);
+    }
+  }
```
Comment (Nana-EC) [inline packages/relay/src/lib/errors/MirrorNodeClientError.ts:31]: nit: new line

File: packages/relay/src/lib/eth.ts [hunk 4]
```diff
@@ -620,13 +627,18 @@ export class EthImpl implements Eth {
...
+        if (result?.type === constants.TYPE_ACCOUNT) {
+            const accountInfo = await this.sdkClient.getAccountInfo(result?.entity.account, EthImpl.ethGetTransactionCount);
+            return EthImpl.numberTo0x(Number(accountInfo.ethereumNonce));  
+        }
+        else if (result?.type === constants.TYPE_CONTRACT) {
+          return EthImpl.numberTo0x(1);
+        }  
+      } catch (e: any) {
+        this.logger.error(e, 'Error raised during getTransactionCount for address %s, block number or tag %s', address, blockNumOrTag);
+        throw e;
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:641]: All returns should be of predefined types where possible
```suggestion
        return predefined.INTERNAL_ERROR;        
```

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Almost there. 1 modification 2 nits

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**